### PR TITLE
Add dbcontext for storing ProxyRoute

### PR DESCRIPTION
**Pull request by Kahbazi**

### CODE REVIEW

**Review by mohsenShakiba — COMMENTED** [review]
why the proxy route configuration is added inside the `ModelBuilderExtensions` class? 
shouldn't it be in a separate EntityTypeConfigurationFile?
Also what is the purpose of creating `IReverseProxyDbContext`? is it because you want to change the implementation at runtime? or to hide DbContext through abstraction?

File: Configuration.EntityFrameworkCore/src/SeoYarp.Configuration.EntityFrameworkCore/Options/ReverseProxyStoreOptions.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        /// </value>
+        public TableConfiguration ProxyRoutes { get; set; } = new TableConfiguration("ProxyRoutes");
+
+        /// <summary>
+        /// Gets or sets the route headers table configuration.
+        /// </summary>
+        /// <value>
+        /// The route headers.
+        /// </value>
+        public TableConfiguration RouteHeaders { get; set; } = new TableConfiguration("RouteHeaders");
```
Comment (mohsenShakiba) [inline Configuration.EntityFrameworkCore/src/SeoYarp.Configuration.EntityFrameworkCore/Options/ReverseProxyStoreOptions.cs:49]: we could also use the singular naming convention
`        public TableConfiguration RouteHeaders { get; set; } = new TableConfiguration(nameof(RouteHeader));
`

**Comment by Kahbazi** [comment]
> why the proxy route configuration is added inside the `ModelBuilderExtensions` class?
> shouldn't it be in a separate EntityTypeConfigurationFile?
> Also what is the purpose of creating `IReverseProxyDbContext`? is it because you want to change the implementation at runtime? or to hide DbContext through abstraction?

I followed the structure from [IdentityServer4.EntityFramework](https://github.com/IdentityServer/IdentityServer4/tree/main/src/EntityFramework.Storage/src)

**Review by mohsenShakiba — COMMENTED** [review]
> > why the proxy route configuration is added inside the `ModelBuilderExtensions` class?
> > shouldn't it be in a separate EntityTypeConfigurationFile?
> > Also what is the purpose of creating `IReverseProxyDbContext`? is it because you want to change the implementation at runtime? or to hide DbContext through abstraction?
> 
> I followed the structure from [IdentityServer4.EntityFramework](https://github.com/IdentityServer/IdentityServer4/tree/main/src/EntityFramework.Storage/src)

How about creating separate `IEntityTypeConfiguration<Entity>` files and add them inside the `ModelBuilderExtension`? won't this make it a bit cleaner? from what I see the ModelBuilderExtension will contain a lot of entity configuration code that belongs to a separate folder. Also I'm basing this on eShopOnContainer project.

**Review by mohsenShakiba — APPROVED** [review]

**Pull request merged**